chain/tests/bitmap_accumulator.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 The Grin Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use self::chain::txhashset::BitmapAccumulator;
+use self::core::core::hash::Hash;
+use self::core::ser::PMMRIndexHashable;
+use bit_vec::BitVec;
+use grin_chain as chain;
+use grin_core as core;
+use grin_util as util;
+
+#[test]
+fn test_bitmap_accumulator() {
+	util::init_test_logger();
+
+	let mut bit_vec = BitVec::from_elem(1024, false);
+	let mut accumulator = BitmapAccumulator::new();
+	assert_eq!(accumulator.root(), Hash::default());
+
+	// 1000... (rebuild from 0, setting [0] true)
+	accumulator.apply(vec![0], vec![0]).unwrap();
+	bit_vec.set(0, true);
+	let expected_bytes = bit_vec.to_bytes();
+	let expected_hash = expected_bytes.hash_with_index(0);
```

Comment:
would't expected_bytes have to be LEN_BYTES long?

Reply:
Are you saying we should just assert this in the test? Or something else?

Reply:
No; I'm puzzled how this test can work, since I expect the UBMT to compute hashes of input length LEN_BYTES at the leaves, while this test hashes expected_bytes whose length I'm not sure how to determine but it appears to be smaller than that, with only bit 0 set.

Reply:
`expected_bytes` comes from `bit_vec` which itself is initialized as 1024 false values - 

https://github.com/mimblewimble/grin/blob/a3bae98b3f306c9d65caa3b2616a0beb2046c5c9/chain/tests/bitmap_accumulator.rs#L27

So `expected_bytes` is 1024 bits in length which corresponds to `LEN_BYTES`.